src/compat/tracker.jl

```diff
@@ -1,105 +1,428 @@
 import .Tracker
-using .Tracker: TrackedReal, TrackedVector, TrackedMatrix, TrackedArray, @grad
+using .Tracker: Tracker,
+                TrackedReal,
+                TrackedVector,
+                TrackedMatrix,
+                TrackedArray,
+                TrackedVecOrMat,
+                @grad,
+                track,
+                data,
+                param
+using LinearAlgebra
+
+Base.prevfloat(r::TrackedReal) = track(prevfloat, r)
+@grad function prevfloat(r::Real)
+    prevfloat(data(r)), Δ -> Δ
+end
+Base.nextfloat(r::TrackedReal) = track(nextfloat, r)
+@grad function nextfloat(r::Real)
+    nextfloat(data(r)), Δ -> Δ
+end
+for i = 0:2, c = Tracker.combinations([:AbstractArray, :TrackedArray, :TrackedReal, :Number], i), f = [:hcat, :vcat]
```

Comment:
Can you explain why this is necessary? Is there an issue with tracking when one of the arguments are `TrackedReal`? 
If so, should this maybe be a fix to https://github.com/FluxML/Tracker.jl/blob/128c29d46d51e182319bbdbae066dc279bdee2af/src/lib/array.jl#L172-L176? Ofc can leave it here as a quickfix for now until that is fixed in Tracker.jl, but maybe at least add a comment about it?

Reply:
True. This needs to be added in Tracker. I just put it here as a quick hack. Tracker was having issues with a single argument `vcat` and other `vcat` methods with `TrackedReal`.